utils/patientmatching/rule_based_patient_matcher.bal

```diff
@@ -0,0 +1,261 @@
+// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerinax/health.fhir.r4;
+import ballerinax/mysql;
+import ballerina/sql;
+import ballerina/http;
+import ballerinax/health.fhir.r4utils.fhirpath as fhirpath;
+
+const RESULT = "result";
+
+# Record to hold Fhirpath rules.
+#
+# + fhirpathArray - array of Fhirpath rules
+public type RulesRecord record {
```

Comment:
Use closed records if `fhirpathArray` is the only field that a `RulesRecord` can contain. Similarly for the other records as well